language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/positioning/resolvers/ServiceScopeResolver.java

```diff
@@ -83,8 +84,14 @@ protected boolean isWithinScopeAfterLastChildNode(Node node, TreeVisitor treeVis
             isLastChildNode = resources.indexOf(node) == (resources.size() - 1);
         }
 
-        return (isLastChildNode
+        boolean isWithinScope =  (isLastChildNode
                 && (curLine < serviceEndLine || (curLine == serviceEndLine && curCol < serviceEndCol))
                 && (nodeEndLine < curLine || (nodeEndLine == curLine && nodeEndCol < curCol)));
```

Comment:
Just a suggestion. We can move this logic to a util function as it is used in few other resolvers.

Reply:
Will need to finalize the scoping after considering the surrounding token analysis as well, until then shall we keep this

Reply:
sure